src/library/scala/collection/mutable/WrappedArray.scala

```diff
@@ -130,7 +130,7 @@ object WrappedArray {
     def update(index: Int, elem: T) { array(index) = elem }
     override def hashCode = MurmurHash3.wrappedArrayHash(array)
     override def equals(that: Any) = that match {
-      case that: ofRef[_] => Arrays.equals(array.asInstanceOf[Array[AnyRef]], that.array.asInstanceOf[Array[AnyRef]])
+      case that: ofRef[_] => array.sameElements(that.array)
```

Comment:
`sameElements` comes from `ArrayOps`, so it boxes both sides which is unnecessary here because you already have two `Seq`s and can call `this.sameElements(that)`. That in turn is essentially the same as `super.equals` (except for an unnecessary `canEqual` check there). Unless there's a noticeable performance improvement from avoiding `canEqual` (but doing a type check and a cast instead) I would  remove the method entirely and let the super implementation (in `IndexedSeqOptimized`) handle everything.

Reply:
`equals` is removed as suggested. Now, it is the same as the branch 2.11.x. Removing `equals` should be a better solution.